Python/jit.c

```diff
@@ -529,7 +540,11 @@ _PyJIT_Compile(_PyExecutorObject *executor, const _PyUOpInstruction trace[], siz
     data += group->data_size;
     assert(code == memory + code_size);
     assert(data == memory + code_size + data_size);
-    if (mark_executable(memory, total_size)) {
+    int status = mark_executable(memory, total_size);
+#ifdef MAP_JIT
+    pthread_jit_write_protect_np(1);
+#endif
+    if (status) {
```

Comment:
This should work, right? Then we don't need `status`:

```suggestion
#ifdef MAP_JIT
    pthread_jit_write_protect_np(1);
#endif
    if (mark_executable(memory, total_size)) {
```

Reply:
I  don't think this is the same thing. if we place `pthread_jit_write_protect_np(1);` is in `mark_executable`, this would be just after the clear cache. The solution you suggested would place it before the clear cache and It could raise an error as it enable the write protection and then the clear cache.

Reply:
I don't think it matters which order they occur in (it's not like the cache invalidation tries to execute the code or anything). I just tried it locally and didn't experience any issues.

Reply:
I tried not to change the previous code logic, but anyway you are right, it works in our internal CI as well. Unless there are other requests, I think this was the last request for change.